### PR TITLE
SM-378 Auto-Join

### PR DESCRIPTION
**Pull request by JCSanPedro**
Mostly contains changes to the contract interfaces to make some queries simpler.

### CODE REVIEW

File: contracts/ProtocolTimeManager.sol [hunk 1]
```diff
@@ -320,8 +320,10 @@ contract ProtocolTimeManager is
         uint256 totalTimeElapsedWithinPeriod = block.timestamp - cycle.start;
 
         uint256 periodDuration = _getPeriodDuration();
+        uint256 periodId = totalTimeElapsedWithinPeriod / periodDuration;
+        uint256 periodStart = cycle.start + periodDuration * periodId;
```
Comment (Nick95550) [inline contracts/ProtocolTimeManager.sol:324]: I think would be better described as 'totalTimeElapsedWithinCycle'

File: contracts/IDirectory.sol [hunk 2]
```diff
@@ -29,9 +29,11 @@ interface IDirectory {
...
-        uint256 rewardCycleId,
-        uint256 stakingPeriodId
+        uint256 cycleId,
+        uint256 periodId
     ) external returns (address);
 
     function joinNextDirectory() external;
+
+    function getDirectoryStake(uint256 cycleId, uint256 periodId, address node) external view returns (uint256);
 }
```
Comment (Nick95550) [inline contracts/IDirectory.sol:39]: We should change the description on the Directory struct to no longer reference epoch

**Review by Nick95550 — APPROVED** [review]
Nothing big just a couple minor comments

File: contracts/ProtocolTimeManager.sol (hunk 1 above)
Comment (JCSanPedro) [reply contracts/ProtocolTimeManager.sol:324]: `totalTimeElapsedWithinCycle` I don't think that accurately reflects what the above value means.

For example, the current time might be `450` seconds into the cycle, but if the period duration is `100s` and we are in the 4th period, then the start of the period is actually `400`.

File: contracts/ProtocolTimeManager.sol [hunk 3]
```diff
@@ -320,8 +320,10 @@ contract ProtocolTimeManager is
         uint256 totalTimeElapsedWithinPeriod = block.timestamp - cycle.start;
 
         uint256 periodDuration = _getPeriodDuration();
+        uint256 periodId = totalTimeElapsedWithinPeriod / periodDuration;
```
Comment (JCSanPedro) [inline contracts/ProtocolTimeManager.sol:323]:
```suggestion
        uint256 periodId = totalTimeElapsedWithinCycle / periodDuration;
```
blah blah blah

**Pull request merged**